# ğŸ§¬ Code for: Unraveling Rare Codon Bias in Actinomycetota: Lineage-Specific and 5'Terminal Enrichment Across 1936 Genomes

This repository contains the analysis code and scripts used in the study:

> **Unraveling Rare Codon Bias in Actinomycetota: Lineage-Specific and 5'Terminal Enrichment Across 1936 Genomes**  
> (Submitted to Genome Biology, October 2025)

---

## ğŸ“˜ Overview
This repository includes the core scripts and workflows used to process, analyze, and visualize data for the manuscript.  
It is currently a **preliminary release (v0.1-preprint)** created to accompany the submission.

---

## ğŸ“ Contents
- `jupyter_notebooks/` â€“ core analysis scripts used for data processing and figure generation  
- `genbank_actinokineospora/` â€“ genbank files for Actinokineospora genera
- `README.md` â€“ current document 

## ğŸ–Šï¸ Note

This repository contains all Jupyter notebooks used for the analysis of **codon usage bias**, **rare codon distribution**, and their enrichment in **biosynthetic gene clusters (BGCs)** across 11 genera of the *Actinomycetota* phylum.  
The primary focus is on the **rare TTA (Leu)** codon, which is known to play a regulatory role in *Streptomyces* and related genera.  
All analyses were also **repeated for the AGA (Arg)** codon using the same pipeline and methodology unless otherwise stated.  

## ğŸ“œ License

This repository is released under the MIT License.
Please refer to the **LICENSE** for details.
