{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "a78244a3-9a38-40c4-b87a-14810dc345d9",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import os\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "bcdd229d-b376-40a3-bdee-2188495179c9",
   "metadata": {},
   "outputs": [],
   "source": [
    "# File with TTA containing Actinokineospora genes \n",
    "file_path = '/results/Actinokineospora/all_first_TTA_Actinokineospora.csv'\n",
    "TTA_df = pd.read_csv(file_path)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "02edf52e-c890-4c35-8e72-5ad563ac2fec",
   "metadata": {},
   "outputs": [],
   "source": [
    "# File with eggnog redults for Actinokineospora\n",
    "file_path = '/input/eggnog/Actinokineospora/concatenated_annotations.csv'\n",
    "concatenated_df = pd.read_csv(file_path)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "7bb4d0d0-823d-4c60-8d6d-b0f9749fdc1d",
   "metadata": {},
   "outputs": [],
   "source": [
    "cog_df = pd.merge(TTA_df, concatenated_df, on = 'Locus_ID', how = 'left')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "1664bc07-2249-4a00-9988-50e6fbf0069a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Locus_ID', 'Product', 'Location', 'UUA Codon Position',\n",
       "       'Relative_Position', 'Gene_Length', 'UUA Codon Count', 'Start_Codon',\n",
       "       'Stop_Codon', 'Strand', 'Sequence', 'Organism', 'GCF_ID',\n",
       "       'seed_ortholog', 'evalue', 'score', 'eggNOG_OGs', 'max_annot_lvl',\n",
       "       'COG_category', 'Description', 'Preferred_name', 'GOs', 'EC', 'KEGG_ko',\n",
       "       'KEGG_Pathway', 'KEGG_Module', 'KEGG_Reaction', 'KEGG_rclass', 'BRITE',\n",
       "       'KEGG_TC', 'CAZy', 'BiGG_Reaction', 'PFAMs', 'Strain'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cog_df.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "1a53f3e2-d949-4e72-a2f5-1d879b238cd7",
   "metadata": {},
   "outputs": [],
   "source": [
    "columns_to_keep = ['Locus_ID', 'Product', 'Location', 'UUA Codon Position',\n",
    "       'Relative_Position', 'Gene_Length', 'UUA Codon Count', 'Start_Codon',\n",
    "       'Stop_Codon', 'Strand', 'Organism', 'GCF_ID',\n",
    "       'COG_category','GOs', 'EC','Strain']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "254e672b-6c15-442a-b584-63ed93a661bb",
   "metadata": {},
   "outputs": [],
   "source": [
    "cog_df = cog_df[columns_to_keep]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "0ac55cc2-8d5e-4139-b019-90283436e8de",
   "metadata": {},
   "outputs": [],
   "source": [
    "cog_df.to_excel('/results/Actinokineospora/cog_TTA_Actinokineospora.xlsx')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d9380b0f-d47c-43fa-911e-34d4b8740142",
   "metadata": {},
   "source": [
    "### Count COGs per strain"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "1b15c7b1-7896-404d-8f48-c6f95498ba3c",
   "metadata": {},
   "outputs": [],
   "source": [
    "# File with summary of how many genes there are for each category per each Actinokineospora genome\n",
    "file_path = '/input/eggnog/Actinokineospora/cog_summary.xlsx'\n",
    "cog_summary_overall = pd.read_excel(file_path, index_col = 'Strain')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "37b9db46-0df6-4a6e-8702-ba265e5b1a8b",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>-</th>\n",
       "      <th>A</th>\n",
       "      <th>BDLTU</th>\n",
       "      <th>BQ</th>\n",
       "      <th>C</th>\n",
       "      <th>CE</th>\n",
       "      <th>CG</th>\n",
       "      <th>CH</th>\n",
       "      <th>CI</th>\n",
       "      <th>CK</th>\n",
       "      <th>...</th>\n",
       "      <th>QT</th>\n",
       "      <th>QU</th>\n",
       "      <th>QV</th>\n",
       "      <th>S</th>\n",
       "      <th>T</th>\n",
       "      <th>U</th>\n",
       "      <th>UW</th>\n",
       "      <th>V</th>\n",
       "      <th>W</th>\n",
       "      <th>Z</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Strain</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>GCF_003182415.1</th>\n",
       "      <td>686</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>252</td>\n",
       "      <td>3</td>\n",
       "      <td>11</td>\n",
       "      <td>22</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>17</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1254</td>\n",
       "      <td>219</td>\n",
       "      <td>40</td>\n",
       "      <td>0</td>\n",
       "      <td>108</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_003663795.1</th>\n",
       "      <td>703</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>246</td>\n",
       "      <td>2</td>\n",
       "      <td>10</td>\n",
       "      <td>26</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1207</td>\n",
       "      <td>280</td>\n",
       "      <td>45</td>\n",
       "      <td>0</td>\n",
       "      <td>106</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_004362515.1</th>\n",
       "      <td>676</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>256</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>13</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>14</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1273</td>\n",
       "      <td>207</td>\n",
       "      <td>50</td>\n",
       "      <td>0</td>\n",
       "      <td>100</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_014648415.1</th>\n",
       "      <td>614</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>284</td>\n",
       "      <td>2</td>\n",
       "      <td>11</td>\n",
       "      <td>18</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>18</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1228</td>\n",
       "      <td>207</td>\n",
       "      <td>46</td>\n",
       "      <td>1</td>\n",
       "      <td>125</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_016907695.1</th>\n",
       "      <td>724</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>246</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>23</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1226</td>\n",
       "      <td>270</td>\n",
       "      <td>44</td>\n",
       "      <td>0</td>\n",
       "      <td>117</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_024171925.1</th>\n",
       "      <td>722</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>252</td>\n",
       "      <td>2</td>\n",
       "      <td>8</td>\n",
       "      <td>25</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>10</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1239</td>\n",
       "      <td>287</td>\n",
       "      <td>49</td>\n",
       "      <td>0</td>\n",
       "      <td>118</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_024760565.1</th>\n",
       "      <td>612</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>270</td>\n",
       "      <td>2</td>\n",
       "      <td>11</td>\n",
       "      <td>19</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>18</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1216</td>\n",
       "      <td>194</td>\n",
       "      <td>45</td>\n",
       "      <td>1</td>\n",
       "      <td>116</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_030268905.1</th>\n",
       "      <td>693</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>246</td>\n",
       "      <td>1</td>\n",
       "      <td>8</td>\n",
       "      <td>32</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1231</td>\n",
       "      <td>287</td>\n",
       "      <td>47</td>\n",
       "      <td>0</td>\n",
       "      <td>115</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_030268945.1</th>\n",
       "      <td>711</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>243</td>\n",
       "      <td>2</td>\n",
       "      <td>10</td>\n",
       "      <td>26</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1195</td>\n",
       "      <td>269</td>\n",
       "      <td>41</td>\n",
       "      <td>0</td>\n",
       "      <td>116</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_900101685.1</th>\n",
       "      <td>703</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>228</td>\n",
       "      <td>2</td>\n",
       "      <td>8</td>\n",
       "      <td>15</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1199</td>\n",
       "      <td>185</td>\n",
       "      <td>32</td>\n",
       "      <td>0</td>\n",
       "      <td>85</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_900111175.1</th>\n",
       "      <td>697</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>251</td>\n",
       "      <td>2</td>\n",
       "      <td>10</td>\n",
       "      <td>22</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1203</td>\n",
       "      <td>284</td>\n",
       "      <td>51</td>\n",
       "      <td>0</td>\n",
       "      <td>116</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>11 rows Ã— 109 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                   -  A  BDLTU  BQ    C  CE  CG  CH  CI  CK  ...  QT  QU  QV  \\\n",
       "Strain                                                       ...               \n",
       "GCF_003182415.1  686  2      1   1  252   3  11  22   1   1  ...  17   0   0   \n",
       "GCF_003663795.1  703  2      1   2  246   2  10  26   1   1  ...   8   1   0   \n",
       "GCF_004362515.1  676  1      1   1  256   2   5  13   1   1  ...  14   0   0   \n",
       "GCF_014648415.1  614  2      1   2  284   2  11  18   2   1  ...  18   0   0   \n",
       "GCF_016907695.1  724  2      1   2  246   2   6  23   1   1  ...   8   1   1   \n",
       "GCF_024171925.1  722  2      1   2  252   2   8  25   1   1  ...  10   1   0   \n",
       "GCF_024760565.1  612  1      1   2  270   2  11  19   2   1  ...  18   0   0   \n",
       "GCF_030268905.1  693  2      1   1  246   1   8  32   1   1  ...   6   1   0   \n",
       "GCF_030268945.1  711  2      1   1  243   2  10  26   1   1  ...   6   1   0   \n",
       "GCF_900101685.1  703  2      1   1  228   2   8  15   1   1  ...  16   0   0   \n",
       "GCF_900111175.1  697  2      1   1  251   2  10  22   1   1  ...   8   1   0   \n",
       "\n",
       "                    S    T   U  UW    V  W  Z  \n",
       "Strain                                         \n",
       "GCF_003182415.1  1254  219  40   0  108  0  0  \n",
       "GCF_003663795.1  1207  280  45   0  106  1  0  \n",
       "GCF_004362515.1  1273  207  50   0  100  0  0  \n",
       "GCF_014648415.1  1228  207  46   1  125  0  0  \n",
       "GCF_016907695.1  1226  270  44   0  117  0  0  \n",
       "GCF_024171925.1  1239  287  49   0  118  0  1  \n",
       "GCF_024760565.1  1216  194  45   1  116  0  0  \n",
       "GCF_030268905.1  1231  287  47   0  115  0  0  \n",
       "GCF_030268945.1  1195  269  41   0  116  0  0  \n",
       "GCF_900101685.1  1199  185  32   0   85  0  0  \n",
       "GCF_900111175.1  1203  284  51   0  116  1  0  \n",
       "\n",
       "[11 rows x 109 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cog_summary_overall"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c8074183-0fa6-4a36-8b7a-637ae0167728",
   "metadata": {},
   "source": [
    "### Count TTA-containing genes per category per genome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "1e6e8246-90ae-4881-a62e-57cae13cb6b2",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/zm/bd67fqqd1cg4ycfsvq5v2x_h0000gp/T/ipykernel_7499/2341941092.py:17: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df[strain_column] = df[strain_column].astype(str).str.strip()\n",
      "/var/folders/zm/bd67fqqd1cg4ycfsvq5v2x_h0000gp/T/ipykernel_7499/2341941092.py:18: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df[cog_column] = df[cog_column].fillna(\"Unknown\")\n"
     ]
    }
   ],
   "source": [
    "def count_cog_categories_per_strain(df, cog_column=\"COG_category\", strain_column=\"GCF_ID\"):\n",
    "    \"\"\"\n",
    "    Count occurrences of each COG category for each strain.\n",
    "    \n",
    "    Parameters:\n",
    "        df (pd.DataFrame): DataFrame containing the data.\n",
    "        cog_column (str): Name of the column containing COG categories.\n",
    "        strain_column (str): Name of the column containing strain names.\n",
    "    \n",
    "    Returns:\n",
    "        pd.DataFrame: Summary DataFrame with counts of each COG category per strain.\n",
    "    \"\"\"\n",
    "    if cog_column not in df.columns or strain_column not in df.columns:\n",
    "        raise ValueError(f\"'{cog_column}' or '{strain_column}' column not found in the DataFrame.\")\n",
    "\n",
    "    # Standardize and fill missing values\n",
    "    df[strain_column] = df[strain_column].astype(str).str.strip()\n",
    "    df[cog_column] = df[cog_column].fillna(\"Unknown\")\n",
    "\n",
    "    # Group by strain and COG category and count occurrences\n",
    "    summary_df = df.groupby([strain_column, cog_column]).size().reset_index(name=\"Count\")\n",
    "\n",
    "    # Pivot table\n",
    "    summary_df = summary_df.pivot(index=strain_column, columns=cog_column, values=\"Count\").fillna(0).astype(int)\n",
    "\n",
    "    # Ensure all strains are included\n",
    "    all_strains = df[strain_column].unique()\n",
    "    summary_df = summary_df.reindex(all_strains, fill_value=0)\n",
    "\n",
    "    return summary_df\n",
    "\n",
    "# Run the function\n",
    "cog_summary_tta = count_cog_categories_per_strain(cog_df, cog_column=\"COG_category\", strain_column=\"GCF_ID\")\n",
    "\n",
    "# Save the summary to a CSV file\n",
    "output_file = '/results/cog_categories_summary_TTA.csv'\n",
    "cog_summary_tta.to_csv(output_file)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "2103585d-8512-4648-8601-ae287666d334",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Keep only columns with single-letter COG categories\n",
    "filtered_df = cog_summary_tta.loc[:, [col for col in cog_summary_tta.columns if len(col) == 1]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "bb868a59-c8f1-4a09-8635-fb283f93c796",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Remove underrepresented COG categories\n",
    "filtered_df = filtered_df.drop(['-', 'S', 'B', 'N', 'A', 'Z', 'W'], axis=1, errors='ignore')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "13bf1238-dbbc-4df1-8b10-6f3fb4ede42f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>COG_category</th>\n",
       "      <th>C</th>\n",
       "      <th>D</th>\n",
       "      <th>E</th>\n",
       "      <th>F</th>\n",
       "      <th>G</th>\n",
       "      <th>H</th>\n",
       "      <th>I</th>\n",
       "      <th>J</th>\n",
       "      <th>K</th>\n",
       "      <th>L</th>\n",
       "      <th>M</th>\n",
       "      <th>O</th>\n",
       "      <th>P</th>\n",
       "      <th>Q</th>\n",
       "      <th>T</th>\n",
       "      <th>U</th>\n",
       "      <th>V</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>GCF_900111175.1</th>\n",
       "      <td>16</td>\n",
       "      <td>2</td>\n",
       "      <td>26</td>\n",
       "      <td>2</td>\n",
       "      <td>27</td>\n",
       "      <td>5</td>\n",
       "      <td>15</td>\n",
       "      <td>10</td>\n",
       "      <td>56</td>\n",
       "      <td>27</td>\n",
       "      <td>20</td>\n",
       "      <td>19</td>\n",
       "      <td>20</td>\n",
       "      <td>39</td>\n",
       "      <td>37</td>\n",
       "      <td>3</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_004362515.1</th>\n",
       "      <td>11</td>\n",
       "      <td>4</td>\n",
       "      <td>24</td>\n",
       "      <td>9</td>\n",
       "      <td>28</td>\n",
       "      <td>10</td>\n",
       "      <td>8</td>\n",
       "      <td>6</td>\n",
       "      <td>46</td>\n",
       "      <td>21</td>\n",
       "      <td>23</td>\n",
       "      <td>12</td>\n",
       "      <td>15</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>2</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_024760565.1</th>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>10</td>\n",
       "      <td>3</td>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "      <td>28</td>\n",
       "      <td>20</td>\n",
       "      <td>10</td>\n",
       "      <td>6</td>\n",
       "      <td>5</td>\n",
       "      <td>7</td>\n",
       "      <td>10</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_900101685.1</th>\n",
       "      <td>10</td>\n",
       "      <td>7</td>\n",
       "      <td>29</td>\n",
       "      <td>2</td>\n",
       "      <td>10</td>\n",
       "      <td>8</td>\n",
       "      <td>11</td>\n",
       "      <td>6</td>\n",
       "      <td>48</td>\n",
       "      <td>32</td>\n",
       "      <td>22</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>23</td>\n",
       "      <td>17</td>\n",
       "      <td>3</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_014648415.1</th>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>4</td>\n",
       "      <td>9</td>\n",
       "      <td>3</td>\n",
       "      <td>7</td>\n",
       "      <td>3</td>\n",
       "      <td>30</td>\n",
       "      <td>13</td>\n",
       "      <td>8</td>\n",
       "      <td>5</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>14</td>\n",
       "      <td>3</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_030268905.1</th>\n",
       "      <td>22</td>\n",
       "      <td>6</td>\n",
       "      <td>38</td>\n",
       "      <td>4</td>\n",
       "      <td>32</td>\n",
       "      <td>12</td>\n",
       "      <td>19</td>\n",
       "      <td>9</td>\n",
       "      <td>77</td>\n",
       "      <td>32</td>\n",
       "      <td>27</td>\n",
       "      <td>20</td>\n",
       "      <td>23</td>\n",
       "      <td>48</td>\n",
       "      <td>38</td>\n",
       "      <td>6</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_003663795.1</th>\n",
       "      <td>15</td>\n",
       "      <td>1</td>\n",
       "      <td>26</td>\n",
       "      <td>5</td>\n",
       "      <td>22</td>\n",
       "      <td>5</td>\n",
       "      <td>18</td>\n",
       "      <td>6</td>\n",
       "      <td>55</td>\n",
       "      <td>26</td>\n",
       "      <td>23</td>\n",
       "      <td>17</td>\n",
       "      <td>18</td>\n",
       "      <td>40</td>\n",
       "      <td>38</td>\n",
       "      <td>2</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_030268945.1</th>\n",
       "      <td>18</td>\n",
       "      <td>3</td>\n",
       "      <td>37</td>\n",
       "      <td>4</td>\n",
       "      <td>33</td>\n",
       "      <td>14</td>\n",
       "      <td>19</td>\n",
       "      <td>11</td>\n",
       "      <td>75</td>\n",
       "      <td>27</td>\n",
       "      <td>28</td>\n",
       "      <td>17</td>\n",
       "      <td>30</td>\n",
       "      <td>48</td>\n",
       "      <td>44</td>\n",
       "      <td>3</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_003182415.1</th>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>17</td>\n",
       "      <td>7</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>11</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_024171925.1</th>\n",
       "      <td>19</td>\n",
       "      <td>6</td>\n",
       "      <td>40</td>\n",
       "      <td>7</td>\n",
       "      <td>23</td>\n",
       "      <td>15</td>\n",
       "      <td>24</td>\n",
       "      <td>7</td>\n",
       "      <td>98</td>\n",
       "      <td>34</td>\n",
       "      <td>40</td>\n",
       "      <td>31</td>\n",
       "      <td>24</td>\n",
       "      <td>42</td>\n",
       "      <td>35</td>\n",
       "      <td>6</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_016907695.1</th>\n",
       "      <td>15</td>\n",
       "      <td>1</td>\n",
       "      <td>30</td>\n",
       "      <td>7</td>\n",
       "      <td>22</td>\n",
       "      <td>11</td>\n",
       "      <td>24</td>\n",
       "      <td>4</td>\n",
       "      <td>68</td>\n",
       "      <td>28</td>\n",
       "      <td>22</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>35</td>\n",
       "      <td>26</td>\n",
       "      <td>5</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "COG_category      C  D   E  F   G   H   I   J   K   L   M   O   P   Q   T  U  \\\n",
       "GCF_ID                                                                         \n",
       "GCF_900111175.1  16  2  26  2  27   5  15  10  56  27  20  19  20  39  37  3   \n",
       "GCF_004362515.1  11  4  24  9  28  10   8   6  46  21  23  12  15  20  20  2   \n",
       "GCF_024760565.1   6  1   4  4  10   3   6   3  28  20  10   6   5   7  10  2   \n",
       "GCF_900101685.1  10  7  29  2  10   8  11   6  48  32  22   7   8  23  17  3   \n",
       "GCF_014648415.1   7  0   6  4   9   3   7   3  30  13   8   5   7   7  14  3   \n",
       "GCF_030268905.1  22  6  38  4  32  12  19   9  77  32  27  20  23  48  38  6   \n",
       "GCF_003663795.1  15  1  26  5  22   5  18   6  55  26  23  17  18  40  38  2   \n",
       "GCF_030268945.1  18  3  37  4  33  14  19  11  75  27  28  17  30  48  44  3   \n",
       "GCF_003182415.1   3  2   4  2   4   1   2   1  17   7   3   4   2  11   4  0   \n",
       "GCF_024171925.1  19  6  40  7  23  15  24   7  98  34  40  31  24  42  35  6   \n",
       "GCF_016907695.1  15  1  30  7  22  11  24   4  68  28  22  20  20  35  26  5   \n",
       "\n",
       "COG_category      V  \n",
       "GCF_ID               \n",
       "GCF_900111175.1  20  \n",
       "GCF_004362515.1  10  \n",
       "GCF_024760565.1   6  \n",
       "GCF_900101685.1  12  \n",
       "GCF_014648415.1   8  \n",
       "GCF_030268905.1  18  \n",
       "GCF_003663795.1  16  \n",
       "GCF_030268945.1  18  \n",
       "GCF_003182415.1   5  \n",
       "GCF_024171925.1  24  \n",
       "GCF_016907695.1  17  "
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "filtered_df"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
