{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "b5f6b241-61e1-4779-af1f-c8b9f5f136f6",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import os\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "import numpy as np\n",
    "import scipy.stats as stats"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "0f8eec71-145b-470c-9bc1-7756d0ca6766",
   "metadata": {},
   "outputs": [],
   "source": [
    "file_path = 'results/Actinokineospora/cog_categories_summary_position_0.csv'\n",
    "cog_tta_1 = pd.read_csv(file_path, index_col = 'Strain')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "b9dc7d2f-5b2c-4e10-ad93-6cc906296a44",
   "metadata": {},
   "outputs": [],
   "source": [
    "cog_tta_1 = cog_tta_1.loc[:, [col for col in cog_tta_1.columns if len(col) == 1]]\n",
    "# Remove underrepresented categories\n",
    "cog_tta_1 = cog_tta_1.drop(['-', 'S', 'B', 'N', 'A', 'Z', 'W'], axis=1, errors='ignore')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "4dbede36-c540-4c50-8b6c-784be2a9a19a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>C</th>\n",
       "      <th>D</th>\n",
       "      <th>E</th>\n",
       "      <th>F</th>\n",
       "      <th>G</th>\n",
       "      <th>H</th>\n",
       "      <th>I</th>\n",
       "      <th>J</th>\n",
       "      <th>K</th>\n",
       "      <th>L</th>\n",
       "      <th>M</th>\n",
       "      <th>O</th>\n",
       "      <th>P</th>\n",
       "      <th>Q</th>\n",
       "      <th>T</th>\n",
       "      <th>U</th>\n",
       "      <th>V</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Strain</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>GCF_003182415.1</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_003663795.1</th>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>4</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>12</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_004362515.1</th>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>2</td>\n",
       "      <td>18</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>22</td>\n",
       "      <td>9</td>\n",
       "      <td>12</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_014648415.1</th>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>5</td>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_016907695.1</th>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>8</td>\n",
       "      <td>9</td>\n",
       "      <td>14</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_024171925.1</th>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>4</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>12</td>\n",
       "      <td>9</td>\n",
       "      <td>14</td>\n",
       "      <td>14</td>\n",
       "      <td>10</td>\n",
       "      <td>12</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_024760565.1</th>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>7</td>\n",
       "      <td>6</td>\n",
       "      <td>5</td>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_030268905.1</th>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>18</td>\n",
       "      <td>4</td>\n",
       "      <td>6</td>\n",
       "      <td>5</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_030268945.1</th>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>16</td>\n",
       "      <td>5</td>\n",
       "      <td>6</td>\n",
       "      <td>4</td>\n",
       "      <td>13</td>\n",
       "      <td>11</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_900101685.1</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>9</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_900111175.1</th>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>5</td>\n",
       "      <td>6</td>\n",
       "      <td>7</td>\n",
       "      <td>9</td>\n",
       "      <td>11</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                 C  D   E  F   G  H  I  J   K  L   M   O   P   Q  T  U  V\n",
       "Strain                                                                   \n",
       "GCF_003182415.1  1  0   2  1   1  0  0  0   5  2   0   1   1   3  0  0  3\n",
       "GCF_003663795.1  5  0   8  0   7  0  3  0   7  4   6   6   6  12  4  0  2\n",
       "GCF_004362515.1  2  0  12  2  18  3  4  1  22  9  12   5   5   7  8  0  4\n",
       "GCF_014648415.1  3  0   2  1   6  1  4  1   6  6   5   3   5   1  5  0  4\n",
       "GCF_016907695.1  2  0  10  1   9  2  5  0  11  6   6   8   9  14  1  0  4\n",
       "GCF_024171925.1  2  1  12  0   8  4  6  1  12  9  14  14  10  12  7  1  4\n",
       "GCF_024760565.1  3  0   1  1   6  0  3  2   7  6   5   3   5   1  3  0  3\n",
       "GCF_030268905.1  6  1  10  0   9  0  1  2  18  4   6   5  11  11  5  0  2\n",
       "GCF_030268945.1  5  0   9  0  10  2  1  2  16  5   6   4  13  11  5  0  2\n",
       "GCF_900101685.1  1  1   9  1   4  1  4  0   7  7   7   1   4   9  5  0  3\n",
       "GCF_900111175.1  8  1   7  0   7  0  1  0   6  5   6   7   9  11  4  0  3"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cog_tta_1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "b75dff6d-8c1c-4092-b048-175625a0e46f",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Read the file for COG categories for TTA containing genes\n",
    "file_path = \"results/Actinokinesospora/cog_categories_summary_TTA.csv\"\n",
    "cog_tta = pd.read_csv(file_path)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "ea490050-b7e4-4810-8a30-ab25e436e457",
   "metadata": {},
   "outputs": [],
   "source": [
    "cog_tta.rename(columns={\"GCF_ID\": \"Strain\"}, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "7293871a-de15-4dcf-a991-05c2126736b8",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Remove \".gbk\" from the \"Strain\" column\n",
    "cog_tta[\"Strain\"] = cog_tta[\"Strain\"].str.replace(\".gbk\", \"\", regex=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "71f30aab-c1d6-4210-a94d-70ecd1aad5a4",
   "metadata": {},
   "outputs": [],
   "source": [
    "cog_tta.set_index('Strain', inplace = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "7956d5da-4eeb-4921-ae38-e858d4f1bb43",
   "metadata": {},
   "outputs": [],
   "source": [
    "cog_tta = cog_tta.loc[:, [col for col in cog_tta_1.columns if len(col) == 1]]\n",
    "# Remove underrepresented categories\n",
    "cog_tta = cog_tta.drop(['-', 'S', 'B', 'N', 'A', 'Z', 'W'], axis=1, errors='ignore')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "7f66759d-69ba-4868-b90c-844edc230f6c",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Read file with summary of COG categories for all genes\n",
    "file_path = '/input/eggnog/Actinokineospora/cog_summary.xlsx'\n",
    "cog_all = pd.read_excel(file_path, index_col = 'Strain')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "884d4f03-6111-418c-bd53-9d86c0a721f1",
   "metadata": {},
   "outputs": [],
   "source": [
    "cog_all = cog_all.loc[:, [col for col in cog_tta_1.columns if len(col) == 1]]\n",
    "cog_all = cog_all.drop(['-', 'S', 'B', 'N', 'A', 'Z', 'W'], axis=1, errors='ignore')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "8e993415-9115-41b9-8a60-b9025a376abb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>C</th>\n",
       "      <th>D</th>\n",
       "      <th>E</th>\n",
       "      <th>F</th>\n",
       "      <th>G</th>\n",
       "      <th>H</th>\n",
       "      <th>I</th>\n",
       "      <th>J</th>\n",
       "      <th>K</th>\n",
       "      <th>L</th>\n",
       "      <th>M</th>\n",
       "      <th>O</th>\n",
       "      <th>P</th>\n",
       "      <th>Q</th>\n",
       "      <th>T</th>\n",
       "      <th>U</th>\n",
       "      <th>V</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Strain</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>GCF_003182415.1</th>\n",
       "      <td>252</td>\n",
       "      <td>65</td>\n",
       "      <td>386</td>\n",
       "      <td>96</td>\n",
       "      <td>203</td>\n",
       "      <td>163</td>\n",
       "      <td>253</td>\n",
       "      <td>187</td>\n",
       "      <td>659</td>\n",
       "      <td>208</td>\n",
       "      <td>239</td>\n",
       "      <td>151</td>\n",
       "      <td>215</td>\n",
       "      <td>256</td>\n",
       "      <td>219</td>\n",
       "      <td>40</td>\n",
       "      <td>108</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_003663795.1</th>\n",
       "      <td>246</td>\n",
       "      <td>63</td>\n",
       "      <td>432</td>\n",
       "      <td>110</td>\n",
       "      <td>188</td>\n",
       "      <td>160</td>\n",
       "      <td>203</td>\n",
       "      <td>204</td>\n",
       "      <td>619</td>\n",
       "      <td>208</td>\n",
       "      <td>230</td>\n",
       "      <td>188</td>\n",
       "      <td>192</td>\n",
       "      <td>216</td>\n",
       "      <td>280</td>\n",
       "      <td>45</td>\n",
       "      <td>106</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_004362515.1</th>\n",
       "      <td>256</td>\n",
       "      <td>61</td>\n",
       "      <td>416</td>\n",
       "      <td>107</td>\n",
       "      <td>275</td>\n",
       "      <td>150</td>\n",
       "      <td>257</td>\n",
       "      <td>190</td>\n",
       "      <td>657</td>\n",
       "      <td>197</td>\n",
       "      <td>258</td>\n",
       "      <td>184</td>\n",
       "      <td>193</td>\n",
       "      <td>224</td>\n",
       "      <td>207</td>\n",
       "      <td>50</td>\n",
       "      <td>100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_014648415.1</th>\n",
       "      <td>284</td>\n",
       "      <td>59</td>\n",
       "      <td>393</td>\n",
       "      <td>101</td>\n",
       "      <td>276</td>\n",
       "      <td>159</td>\n",
       "      <td>218</td>\n",
       "      <td>208</td>\n",
       "      <td>634</td>\n",
       "      <td>190</td>\n",
       "      <td>219</td>\n",
       "      <td>182</td>\n",
       "      <td>198</td>\n",
       "      <td>232</td>\n",
       "      <td>207</td>\n",
       "      <td>46</td>\n",
       "      <td>125</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_016907695.1</th>\n",
       "      <td>246</td>\n",
       "      <td>57</td>\n",
       "      <td>446</td>\n",
       "      <td>107</td>\n",
       "      <td>187</td>\n",
       "      <td>164</td>\n",
       "      <td>201</td>\n",
       "      <td>205</td>\n",
       "      <td>626</td>\n",
       "      <td>215</td>\n",
       "      <td>238</td>\n",
       "      <td>201</td>\n",
       "      <td>196</td>\n",
       "      <td>213</td>\n",
       "      <td>270</td>\n",
       "      <td>44</td>\n",
       "      <td>117</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_024171925.1</th>\n",
       "      <td>252</td>\n",
       "      <td>65</td>\n",
       "      <td>434</td>\n",
       "      <td>111</td>\n",
       "      <td>185</td>\n",
       "      <td>159</td>\n",
       "      <td>219</td>\n",
       "      <td>211</td>\n",
       "      <td>669</td>\n",
       "      <td>198</td>\n",
       "      <td>257</td>\n",
       "      <td>203</td>\n",
       "      <td>218</td>\n",
       "      <td>245</td>\n",
       "      <td>287</td>\n",
       "      <td>49</td>\n",
       "      <td>118</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_024760565.1</th>\n",
       "      <td>270</td>\n",
       "      <td>59</td>\n",
       "      <td>379</td>\n",
       "      <td>96</td>\n",
       "      <td>268</td>\n",
       "      <td>154</td>\n",
       "      <td>221</td>\n",
       "      <td>206</td>\n",
       "      <td>640</td>\n",
       "      <td>208</td>\n",
       "      <td>217</td>\n",
       "      <td>180</td>\n",
       "      <td>192</td>\n",
       "      <td>224</td>\n",
       "      <td>194</td>\n",
       "      <td>45</td>\n",
       "      <td>116</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_030268905.1</th>\n",
       "      <td>246</td>\n",
       "      <td>67</td>\n",
       "      <td>414</td>\n",
       "      <td>111</td>\n",
       "      <td>197</td>\n",
       "      <td>153</td>\n",
       "      <td>208</td>\n",
       "      <td>203</td>\n",
       "      <td>641</td>\n",
       "      <td>207</td>\n",
       "      <td>242</td>\n",
       "      <td>191</td>\n",
       "      <td>207</td>\n",
       "      <td>218</td>\n",
       "      <td>287</td>\n",
       "      <td>47</td>\n",
       "      <td>115</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_030268945.1</th>\n",
       "      <td>243</td>\n",
       "      <td>57</td>\n",
       "      <td>418</td>\n",
       "      <td>107</td>\n",
       "      <td>200</td>\n",
       "      <td>154</td>\n",
       "      <td>207</td>\n",
       "      <td>202</td>\n",
       "      <td>617</td>\n",
       "      <td>196</td>\n",
       "      <td>244</td>\n",
       "      <td>195</td>\n",
       "      <td>204</td>\n",
       "      <td>218</td>\n",
       "      <td>269</td>\n",
       "      <td>41</td>\n",
       "      <td>116</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_900101685.1</th>\n",
       "      <td>228</td>\n",
       "      <td>62</td>\n",
       "      <td>356</td>\n",
       "      <td>92</td>\n",
       "      <td>176</td>\n",
       "      <td>150</td>\n",
       "      <td>229</td>\n",
       "      <td>203</td>\n",
       "      <td>536</td>\n",
       "      <td>250</td>\n",
       "      <td>226</td>\n",
       "      <td>150</td>\n",
       "      <td>166</td>\n",
       "      <td>200</td>\n",
       "      <td>185</td>\n",
       "      <td>32</td>\n",
       "      <td>85</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_900111175.1</th>\n",
       "      <td>251</td>\n",
       "      <td>69</td>\n",
       "      <td>423</td>\n",
       "      <td>110</td>\n",
       "      <td>194</td>\n",
       "      <td>164</td>\n",
       "      <td>199</td>\n",
       "      <td>199</td>\n",
       "      <td>623</td>\n",
       "      <td>220</td>\n",
       "      <td>237</td>\n",
       "      <td>193</td>\n",
       "      <td>198</td>\n",
       "      <td>214</td>\n",
       "      <td>284</td>\n",
       "      <td>51</td>\n",
       "      <td>116</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   C   D    E    F    G    H    I    J    K    L    M    O  \\\n",
       "Strain                                                                       \n",
       "GCF_003182415.1  252  65  386   96  203  163  253  187  659  208  239  151   \n",
       "GCF_003663795.1  246  63  432  110  188  160  203  204  619  208  230  188   \n",
       "GCF_004362515.1  256  61  416  107  275  150  257  190  657  197  258  184   \n",
       "GCF_014648415.1  284  59  393  101  276  159  218  208  634  190  219  182   \n",
       "GCF_016907695.1  246  57  446  107  187  164  201  205  626  215  238  201   \n",
       "GCF_024171925.1  252  65  434  111  185  159  219  211  669  198  257  203   \n",
       "GCF_024760565.1  270  59  379   96  268  154  221  206  640  208  217  180   \n",
       "GCF_030268905.1  246  67  414  111  197  153  208  203  641  207  242  191   \n",
       "GCF_030268945.1  243  57  418  107  200  154  207  202  617  196  244  195   \n",
       "GCF_900101685.1  228  62  356   92  176  150  229  203  536  250  226  150   \n",
       "GCF_900111175.1  251  69  423  110  194  164  199  199  623  220  237  193   \n",
       "\n",
       "                   P    Q    T   U    V  \n",
       "Strain                                   \n",
       "GCF_003182415.1  215  256  219  40  108  \n",
       "GCF_003663795.1  192  216  280  45  106  \n",
       "GCF_004362515.1  193  224  207  50  100  \n",
       "GCF_014648415.1  198  232  207  46  125  \n",
       "GCF_016907695.1  196  213  270  44  117  \n",
       "GCF_024171925.1  218  245  287  49  118  \n",
       "GCF_024760565.1  192  224  194  45  116  \n",
       "GCF_030268905.1  207  218  287  47  115  \n",
       "GCF_030268945.1  204  218  269  41  116  \n",
       "GCF_900101685.1  166  200  185  32   85  \n",
       "GCF_900111175.1  198  214  284  51  116  "
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cog_all"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "5d4211b6-05e2-4418-a51f-6a0f680dd2ba",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Function to perform hypergeometric test for each strain and COG category\n",
    "def perform_hypergeometric_test(cog_tta_1, cog_all, cog_tta):\n",
    "    results = []\n",
    "\n",
    "    for strain in cog_tta_1.index:\n",
    "        if strain in cog_all.index and strain in cog_tta.index:\n",
    "            for cog in cog_tta_1.columns:\n",
    "                # Get values for hypergeometric test\n",
    "                k = cog_tta_1.loc[strain, cog]  # Successes in sample (TTA in bin 1)\n",
    "                K = cog_tta.loc[strain, cog]  # Total TTA in COG category\n",
    "                n = cog_all.loc[strain, cog]  # Sample size (all genes in COG category)\n",
    "                N = cog_all.loc[strain].sum()  # Population size (all genes across all COGs)\n",
    "\n",
    "                # Perform hypergeometric test only if valid numbers\n",
    "                if k > 0 and K > 0 and n > 0 and N > 0:\n",
    "                    p_value = stats.hypergeom.sf(k-1, N, K, n)\n",
    "                else:\n",
    "                    p_value = 1.0  # Default to non-significant if values are invalid\n",
    "\n",
    "                # Store results\n",
    "                results.append({\n",
    "                    \"Strain\": strain,\n",
    "                    \"COG\": cog,\n",
    "                    \"k (TTA in bin 1)\": k,\n",
    "                    \"K (Total TTA in COG)\": K,\n",
    "                    \"n (Total in COG)\": n,\n",
    "                    \"N (Total genes)\": N,\n",
    "                    \"p-value\": p_value\n",
    "                })\n",
    "\n",
    "    # Convert to DataFrame\n",
    "    results_df = pd.DataFrame(results)\n",
    "\n",
    "    return results_df\n",
    "\n",
    "# Assuming the dataframes are already loaded as cog_tta_1, cog_all, and cog_tta\n",
    "hypergeometric_results = perform_hypergeometric_test(cog_tta_1, cog_all, cog_tta)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "088d4a98-acbe-4135-a298-0d10ff64ecde",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Strain</th>\n",
       "      <th>COG</th>\n",
       "      <th>k (TTA in bin 1)</th>\n",
       "      <th>K (Total TTA in COG)</th>\n",
       "      <th>n (Total in COG)</th>\n",
       "      <th>N (Total genes)</th>\n",
       "      <th>p-value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCF_003182415.1</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>252</td>\n",
       "      <td>3700</td>\n",
       "      <td>0.190772</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCF_003182415.1</td>\n",
       "      <td>D</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>65</td>\n",
       "      <td>3700</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCF_003182415.1</td>\n",
       "      <td>E</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td>386</td>\n",
       "      <td>3700</td>\n",
       "      <td>0.056480</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCF_003182415.1</td>\n",
       "      <td>F</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>96</td>\n",
       "      <td>3700</td>\n",
       "      <td>0.051226</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCF_003182415.1</td>\n",
       "      <td>G</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>203</td>\n",
       "      <td>3700</td>\n",
       "      <td>0.202125</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            Strain COG  k (TTA in bin 1)  K (Total TTA in COG)  \\\n",
       "0  GCF_003182415.1   C                 1                     3   \n",
       "1  GCF_003182415.1   D                 0                     2   \n",
       "2  GCF_003182415.1   E                 2                     4   \n",
       "3  GCF_003182415.1   F                 1                     2   \n",
       "4  GCF_003182415.1   G                 1                     4   \n",
       "\n",
       "   n (Total in COG)  N (Total genes)   p-value  \n",
       "0               252             3700  0.190772  \n",
       "1                65             3700  1.000000  \n",
       "2               386             3700  0.056480  \n",
       "3                96             3700  0.051226  \n",
       "4               203             3700  0.202125  "
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hypergeometric_results.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "2741ec1f-2764-46b6-872d-b869d5336557",
   "metadata": {},
   "outputs": [],
   "source": [
    "hypergeometric_results.to_excel('/results/Actinokineospora/Actinokineospora_enrichment_pos_0.xlsx')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "f85078e4-7869-45ce-8b03-9831f6c17746",
   "metadata": {},
   "outputs": [],
   "source": [
    "filtered_df = hypergeometric_results[hypergeometric_results[\"p-value\"] < 0.05]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "97054098-ab70-478b-982a-25a36aca5474",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Count how many strains show enrichment (p-value < 0.05) for each category\n",
    "enrichment_summary = hypergeometric_results[hypergeometric_results[\"p-value\"] < 0.05].groupby(\"COG\")[\"Strain\"].nunique().reset_index()\n",
    "\n",
    "# Rename columns for clarity\n",
    "enrichment_summary.columns = [\"Category\", \"Number of Enriched Strains\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "c4f46672-c8e3-4baf-81b4-d4e2a7bf2c6c",
   "metadata": {},
   "outputs": [],
   "source": [
    "cog_all = cog_all.reset_index()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "b5717257-9bed-4857-9125-093ef75d192d",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Get the total number of strains from cog_summary_overall\n",
    "total_strains = cog_all[\"Strain\"].nunique()\n",
    "\n",
    "# Calculate the percentage of strains in which each category is enriched\n",
    "enrichment_summary[\"Percentage Enriched\"] = (enrichment_summary[\"Number of Enriched Strains\"] / total_strains) * 100"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "4b0bd506-5001-412c-9a60-8cf7bcafb82b",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Map full COG category names\n",
    "enrichment_summary[\"Category\"] = enrichment_summary[\"Category\"].map(cog_full_names)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "585cefe9-0b24-407d-897d-e619bc52863b",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Sorting the enrichment_summary dataframe by the \"Percentage Enriched\" column in descending order\n",
    "enrichment_summary = enrichment_summary.sort_values(by=\"Category\", ascending=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "c3a6627d-b7ca-4500-91a9-fd01450dfd9a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Category</th>\n",
       "      <th>Number of Enriched Strains</th>\n",
       "      <th>Percentage Enriched</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Amino acid transport and metabolism</td>\n",
       "      <td>8</td>\n",
       "      <td>72.727273</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Carbohydrate transport and metabolism</td>\n",
       "      <td>10</td>\n",
       "      <td>90.909091</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Cell cycle control</td>\n",
       "      <td>1</td>\n",
       "      <td>9.090909</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Cell wall/membrane/envelope biogenesis</td>\n",
       "      <td>10</td>\n",
       "      <td>90.909091</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Coenzyme transport and metabolism</td>\n",
       "      <td>2</td>\n",
       "      <td>18.181818</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Defense mechanisms</td>\n",
       "      <td>8</td>\n",
       "      <td>72.727273</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Energy production and conversion</td>\n",
       "      <td>6</td>\n",
       "      <td>54.545455</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Inorganic ion transport and metabolism</td>\n",
       "      <td>10</td>\n",
       "      <td>90.909091</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Lipid transport and metabolism</td>\n",
       "      <td>6</td>\n",
       "      <td>54.545455</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Nucleotide transport and metabolism</td>\n",
       "      <td>1</td>\n",
       "      <td>9.090909</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Posttranslational modifications</td>\n",
       "      <td>9</td>\n",
       "      <td>81.818182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Replication, recombination, and repair</td>\n",
       "      <td>8</td>\n",
       "      <td>72.727273</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Secondary metabolite biosynthesis</td>\n",
       "      <td>9</td>\n",
       "      <td>81.818182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Signal transduction mechanisms</td>\n",
       "      <td>5</td>\n",
       "      <td>45.454545</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Transcription</td>\n",
       "      <td>1</td>\n",
       "      <td>9.090909</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Translation, ribosomal structure, and biogenesis</td>\n",
       "      <td>1</td>\n",
       "      <td>9.090909</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                            Category  \\\n",
       "2                Amino acid transport and metabolism   \n",
       "4              Carbohydrate transport and metabolism   \n",
       "1                                 Cell cycle control   \n",
       "10            Cell wall/membrane/envelope biogenesis   \n",
       "5                  Coenzyme transport and metabolism   \n",
       "15                                Defense mechanisms   \n",
       "0                   Energy production and conversion   \n",
       "12            Inorganic ion transport and metabolism   \n",
       "6                     Lipid transport and metabolism   \n",
       "3                Nucleotide transport and metabolism   \n",
       "11                   Posttranslational modifications   \n",
       "9             Replication, recombination, and repair   \n",
       "13                 Secondary metabolite biosynthesis   \n",
       "14                    Signal transduction mechanisms   \n",
       "8                                      Transcription   \n",
       "7   Translation, ribosomal structure, and biogenesis   \n",
       "\n",
       "    Number of Enriched Strains  Percentage Enriched  \n",
       "2                            8            72.727273  \n",
       "4                           10            90.909091  \n",
       "1                            1             9.090909  \n",
       "10                          10            90.909091  \n",
       "5                            2            18.181818  \n",
       "15                           8            72.727273  \n",
       "0                            6            54.545455  \n",
       "12                          10            90.909091  \n",
       "6                            6            54.545455  \n",
       "3                            1             9.090909  \n",
       "11                           9            81.818182  \n",
       "9                            8            72.727273  \n",
       "13                           9            81.818182  \n",
       "14                           5            45.454545  \n",
       "8                            1             9.090909  \n",
       "7                            1             9.090909  "
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "enrichment_summary"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "1c298fca-c940-4de7-82e9-28976dcb6866",
   "metadata": {},
   "outputs": [],
   "source": [
    "enrichment_summary.to_excel('/results/Actinokineospora_enrichment_summary_pos_0_TTA.xlsx')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
